### DISCUSSION TITLE
Transition Problem when the product is added

### DISCUSSION THREAD

**Discussion by bdoroshev** [EvgeniyMukhamedjanov/liquid-ajax-cart#29] [Q&A] [answered]
This is just a great solution, thank you Eugene, a star from me. Transition on cart open works in all scenarios, but for some reason, it does not work when I add a product, what could this be about?  https://www.zendxb.com/ password: thupeu

**Comment by EvgeniyMukhamedjanov** [comment, marked as answer]
Hey, thank you for the star!
The transition doesn't work because it is set on the element (`.cart-drawer`) with the `data-ajax-cart-section` attribute that is being completely replaced with the updated element after a user adds a product to the cart.
So the solution is to separate the `data-ajax-cart-section` and the element with transition (`.cart-drawer`):
```
<div class="cart-drawer">
  <div class="cart-drawer__wrapper" data-ajax-cart-section>
    <!-- ... -->
  </div>
</div>
```

**Reply by bdoroshev** [reply]
Thank you so much, you are a genius, for sure!
